bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/tools/wizard/ToolsManagerWizardDialog.java

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * Copyright 2022 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.ui.tools.wizard;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.jface.wizard.IWizard;
+import org.eclipse.jface.wizard.IWizardPage;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.swt.widgets.Shell;
+
+import com.espressif.idf.core.IDFEnvironmentVariables;
+import com.espressif.idf.ui.UIPlugin;
+import com.espressif.idf.ui.tools.Messages;
+import com.espressif.idf.ui.tools.wizard.pages.IToolsWizardPage;
+import com.espressif.idf.ui.tools.wizard.pages.ManageToolsInstallationWizardPage;
+
+/**
+ * Tools manager wizard dialog class to extend some basic methods
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class ToolsManagerWizardDialog extends WizardDialog
+{
+	private Map<String, String> existingVarMap;
+	private Map<IWizardPage, Point> wizardPagePointMap;
+	private boolean exisitngInstallPreferencesStatus;
+
+	public ToolsManagerWizardDialog(Shell parentShell, IWizard newWizard, Map<String, String> existingVarMap, boolean exisitngInstallPreferencesStatus)
+	{
+		super(parentShell, newWizard);
+		this.existingVarMap = existingVarMap;
+		wizardPagePointMap = new HashMap<IWizardPage, Point>();
+		this.exisitngInstallPreferencesStatus = exisitngInstallPreferencesStatus;
+	}
+
+	@Override
+	public void showPage(IWizardPage page)
+	{
+		super.showPage(page);
+		updateSize();
+	}
+
+	@Override
+	protected void cancelPressed()
+	{
+		MessageBox messageBox = new MessageBox(getParentShell(), SWT.ICON_WARNING | SWT.YES | SWT.NO);
+		messageBox.setText(Messages.BtnCancel);
```

Comment:
Prefer MessageDialog from Jface

Reply:
this message box is removed as per the comment for eclipse UX guidelines of another dialog from a dialog